js-envs/src/main/scala/org/scalajs/jsenv/phantomjs/PhantomJSEnv.scala

```diff
@@ -249,9 +253,9 @@ class PhantomJSEnv(
       loop("")
     }
 
-    private def receiveFrag(): String = {
+    private def receiveFrag(deadline: OptDeadline): String = {
       while (recvBuf.isEmpty && !mgr.isClosed)
-        wait()
+        wait(deadline.millisLeft)
```

Comment:
This takes a Long, and not an Int. You are converting the millis toInt and then toLong...
Also, you need to check if the deadline expired in the outer loop. Otherwise it will loop anyways (IIUC `wait` does not throw, if it wakes up because of time).